### DISCUSSION TITLE
ZeroWidthString component why is \uFEFF sometimes rendered and sometimes not

### DISCUSSION THREAD

**Discussion by bmingles** [ianstormtaylor/slate#5751] [Q&A]
In the `ZeroWidthString` component, there is a conditionally rendered `\uFEFF` character:

```typescript
{!(IS_ANDROID || IS_IOS) || !isLineBreak ? '\uFEFF' : null}
```

https://github.com/ianstormtaylor/slate/blob/main/packages/slate-react/src/components/string.tsx#L132

I'd like to better understand why / when it is needed vs not.

**Comment by pex** [comment]
Same, especially since it leads to hydration errors on mobile devices in SSR scenarios.

**Comment by bmingles** [comment]
Related issue https://github.com/ianstormtaylor/slate/issues/5760
